lib/src/json_schema/json_schema.dart

```diff
@@ -58,170 +64,541 @@ class JsonSchema {
+
+  /// A [List<JsonSchema>] which the value must conform to all of.
   List<JsonSchema> get allOf => _allOf;
+
+  /// A [List<JsonSchema>] which the value must conform to at least one of.
   List<JsonSchema> get anyOf => _anyOf;
+
+  /// A [List<JsonSchema>] which the value must conform to at least one of.
   List<JsonSchema> get oneOf => _oneOf;
+
+  /// A [JsonSchema] which the value must NOT be.
   JsonSchema get notSchema => _notSchema;
-  Map<String, JsonSchema> get definitions => _definitions;
-  Uri get id => _id;
+
+  /// Ref to the URI of the [JsonSchema].
   String get ref => _ref;
-  String get description => _description;
+
+  /// Title of the [JsonSchema].
   String get title => _title;
+
+  /// TODO
   List<SchemaType> get schemaTypeList => _schemaTypeList;
 
-  /// To match all items to a schema
+  // --------------------------------------------------------------------------
+  // Schema List Item Related Getters
+  // --------------------------------------------------------------------------
+
+  /// Single [JsonSchema] sub items of this [JsonSchema] must conform to.
   JsonSchema get items => _items;
 
-  /// To match each item in array to a schema
+  /// Ordered list of [JsonSchema] which the value of the same index must conform to.
   List<JsonSchema> get itemsList => _itemsList;
-  dynamic get additionalItems => _additionalItems;
+
+  /// Whether additional items are allowed.
+  /* union bool | Map */ dynamic get additionalItems => _additionalItems;
+
+  /// The maximum number of items allowed.
   int get maxItems => _maxItems;
+
+  /// The minimum number of items allowed.
   int get minItems => _minItems;
+
+  /// Whether the items in the list must be unique.
   bool get uniqueItems => _uniqueItems;
-  List<String> get requiredProperties => _requiredProperties;
-  int get maxProperties => _maxProperties;
-  int get minProperties => _minProperties;
+
+  // --------------------------------------------------------------------------
+  // Schema Sub-Property Related Getters
+  // --------------------------------------------------------------------------
+
+  /// Map of [JsonSchema]s for properties, by [String] key.
   Map<String, JsonSchema> get properties => _properties;
+
+  /// Whether additional properties, other than those specified, are allowed.
   bool get additionalProperties => _additionalProperties;
+
+  /// [JsonSchema] that additional properties must conform to.
   JsonSchema get additionalPropertiesSchema => _additionalPropertiesSchema;
+
+  /// The maximum number of properties allowed.
+  int get maxProperties => _maxProperties;
+
+  /// The minimum number of properties allowed.
+  int get minProperties => _minProperties;
+
+  /// Map of [JsonSchema]s for properties, based on [RegExp]s keys.
   Map<RegExp, JsonSchema> get patternProperties => _patternProperties;
-  Map<String, JsonSchema> get schemaDependencies => _schemaDependencies;
+
+  /// TODO: write an informative comment for this.
```

Comment:
Do you want to resolve this TODO before merging this PR?

Reply:
nope, don't really know what this does :/...